src/test/java/org/alfresco/rest/api/search/SearchMapperTests.java

```diff
@@ -288,9 +296,34 @@ public void fromDefaults() throws Exception
     }
 
     @Test
-    public void validateInclude() throws Exception
+    public void searchParameterIncludesValidation_validInclude_shouldReturnSuccessfulValidation()
     {
+        searchMapper.validateInclude(Arrays.asList(PARAM_INCLUDE_ALLOWABLEOPERATIONS,PARAM_INCLUDE_ASPECTNAMES,
+                PARAM_INCLUDE_ISLINK, PARAM_INCLUDE_PATH, PARAM_INCLUDE_PROPERTIES,
+                PARAM_INCLUDE_ASSOCIATION, PARAM_INCLUDE_ISLOCKED, PARAM_INCLUDE_PERMISSIONS));
+    }
 
+    
+    @Test
+    public void searchParameterIncludesValidation_invalidInclude_shouldThrowInvalidArgumentException() throws Exception
```

Comment:
Is this new test really needed? Is it not already covered with the existing validateInclude method? And maybe the same existing method could have been extended to cover the newly added parameters?

Reply:
I usually prefer very verbose tests (with very verbose names to easily identify what is broken when it is broken), covering the possible test cases.
I think there was a validateInclude test method indeed but that just tried to validate a "sausage"  include and checking the failure.
I didn't touch it but I don't have any hard position on that, if you prefer to have just that test to cover the different test cases I will happily do the change :)

Reply:
In this case the name was too verbose for me :) I would have just added the new parameters ("isLocked" and "permissions") to the list of valid permissions and the negative case was covered already. That's a personal opinion though. Normally there is nothing wrong with having more (specific) tests. I don't mind leaving the tests as they are.
I'd like to understand if the parameters were left out for a good reason (in which case the docs should be updated) or if they were just forgotten.

Reply:
Great!
In relation to that doubt you find more information on our investigations in the Jira ticket [1].
From the code history investigations it may be possible they were added but forgotten to be added to the valid include option list.

We have added 3 integration tests in tas to verify that[2].
Integration tests were failing on current Alfresco but succeeding on the fix branch.
At this stage we are not aware of other methods to validate this, but we are open to listen.


[1] https://issues.alfresco.com/jira/browse/MNT-19952
[2] https://git.alfresco.com/tas/alfresco-tas-restapi-test/merge_requests/86